table/tables/tables.go

```diff
@@ -1456,9 +1456,6 @@ func GetColDefaultValue(ctx sessionctx.Context, col *table.Column, defaultVals [
 	if col.GetOriginDefaultValue() == nil && mysql.HasNotNullFlag(col.GetFlag()) {
 		return colVal, errors.New("Miss column")
 	}
-	if col.State != model.StatePublic {
```

Comment:
Why remove it? Is it safe?

Reply:
Consider the case:

```sql
create table t (a tinyint);
insert into t set a = 10 ;
alter table t add column b int not null, change column a c char(5) first;
```

In this ALTER TABLE statement, we add a **NOT NULL** column first, and then do the lossy column type change. In the phase of backfilling, the schema state of column `b` is still "write-reorg". If we keep this code, a null value is assigned to the column `b`, which violates the not-null constraint. Also see https://github.com/tangenta/tidb/issues/94.

I don't see if there are any benefits to get a null value for the middle-state column... so let it be the real default value.

No tests are broken by this change.